### PR TITLE
Refactored VCF module 

### PR DESCRIPTION
**Pull request by roberthorlings**
Refactored the VCF module to be in line with other modules and support the all_data projection

### CODE REVIEW

**Comment by cataphract** [comment]
The core-api and core-db branch names should match so that travis can find them.

File: transmart-core-db-tests/test/integration/org/transmartproject/db/dataquery/highdim/vcf/VcfEndToEndRetrievalTests.groovy [hunk 1]
```diff
@@ -253,9 +235,9 @@ class VcfEndToEndRetrievalTests {
...
 
         assertThat map, hasKey(
-                hasProperty('dataTypeName', equalTo('cohortMAF')))
+                hasProperty('dataTypeName', equalTo('vcf')))
 
-        def entry = map.entrySet().find { it.key.dataTypeName == 'cohortMAF' }
+        def entry = map.entrySet().find { it.key.dataTypeName == 'vcf' }
 
         assertThat entry.value, allOf(
                 hasSize(greaterThan(0)),
```
Comment (cataphract) [inline transmart-core-db-tests/test/integration/org/transmartproject/db/dataquery/highdim/vcf/VcfEndToEndRetrievalTests.groovy:250]: I see here no tests for the two new projections and no test for the fact that the cells in the cohort projection contain the variant.

Comment (cataphract) [inline transmart-core-db-tests/test/integration/org/transmartproject/db/dataquery/highdim/vcf/VcfEndToEndRetrievalTests.groovy:250]: I think the export projetions have their own test hierarchies, so i'd be probably be easier to use its own class for that one.

File: transmart-core-db-tests/test/integration/org/transmartproject/db/dataquery/highdim/vcf/VcfEndToEndRetrievalTests.groovy [hunk 2]
```diff
@@ -211,31 +205,19 @@ class VcfEndToEndRetrievalTests {
...
         )
         assertThat resultList, hasItem(
                 allOf(
                         hasProperty('position', equalTo(3L)),
-                        hasProperty('mafAllele', equalTo('C')),
-                        hasProperty('maf', closeTo(0.5 as Double, 0.1 as Double)),
-                        hasProperty('genomicVariantTypes', contains(org.transmartproject.core.dataquery.highdim.vcf.GenomicVariantType.SNP,
-                                org.transmartproject.core.dataquery.highdim.vcf.GenomicVariantType.SNP)),
-                        hasProperty('alternativeAlleles', containsInAnyOrder('C', 'T')),
-                        hasProperty('additionalInfo', allOf(
```
Comment (cataphract) [inline transmart-core-db-tests/test/integration/org/transmartproject/db/dataquery/highdim/vcf/VcfEndToEndRetrievalTests.groovy:229]: What happened to the other fields in (the now-called) `VcfDataRow`? The ones that are part of the contract (i.e. the ones in `VcfValues` and `RegionRow`) should be part of the assertion.

File: src/groovy/org/transmartproject/db/dataquery/highdim/vcf/VcfModule.groovy [hunk 3]
```diff
@@ -132,12 +148,37 @@ class CohortMAFModule extends AbstractHighDimensionDataTypeModule {
...
-                finalizeGroup:         { List collectedEntries -> /* list of all the results belonging to a group defined by inSameGroup */
+                inSameGroup:           { a, b -> a[0].chr == b[0].chr && a[0].pos == b[0].pos },
+                finalizeGroup:         { List list -> /* list of all the results belonging to a group defined by inSameGroup */
                     /* list of arrays with one element: a map */
                     /* we may have nulls if allowMissingAssays is true,
-                     *, but we're guaranteed to have at least one non-null */
-                    projection.doWithResult(collectedEntries)
+                    *, but we're guaranteed to have at least one non-null */
+                    /* we may have nulls if allowMissingAssays is true,
+                     * but we're guaranteed to have at least one non-null */
```
Comment (cataphract) [inline src/groovy/org/transmartproject/db/dataquery/highdim/vcf/VcfModule.groovy]: Seems to be duped.

**Comment by rnugraha** [comment]
@roberthorlings : This changes have impacted VCF visualization in transmart-mydas. Need your advice to refactor necessary classes.

**Pull request merged**